### PR TITLE
feat(connector-iroha): update-iroha-js

### PR DESCRIPTION
**Pull request by yashrajdesai**
cactus-iroha plugin updated with the latest iroha-js library.

### CODE REVIEW

File: packages/cactus-plugin-ledger-connector-iroha/package.json
Comment (baziorek) [inline packages/cactus-plugin-ledger-connector-iroha/package.json:66]: Isn't conflict between those two? Shouldn't one of them be removed?

**Review by baziorek — COMMENTED** [review]
It is really good that You are trying to update iroha in Cactus's plugin. Unfortunately with the update we need to wait for Iroha-Core team to create another release. Currently Iroha has release 1.5, and 1.6 LTS release is in progress. Unfortunately last release of iroha-javascript is 1.3.0. So before updating iroha in the Plugin we have to wait for the next release of iroha-javascript library.

You have made PR to iroha-javascript: https://github.com/hyperledger/iroha-javascript/pull/125, after merging its we should ask Iroha-core team to create release. After that we would be able to complete PR https://github.com/hyperledger/cactus/pull/2166/.

**Comment by baziorek** [comment]
BTW. Remember to update Cactus-Iroha AIO: https://github.com/hyperledger/cactus/blob/main/tools/docker/iroha-all-in-one/Dockerfile

**Comment by izuru0** [comment]
> It is really good that You are trying to update iroha in Cactus's plugin. Unfortunately with the update we need to wait for Iroha-Core team to create another release. Currently Iroha has release 1.5, and 1.6 LTS release is in progress. Unfortunately last release of iroha-javascript is 1.3.0. So before updating iroha in the Plugin we have to wait for the next release of iroha-javascript library.
I will review after iroha-javascript is updated.

File: packages/cactus-plugin-ledger-connector-iroha/package.json
Comment (yashrajdesai) [reply packages/cactus-plugin-ledger-connector-iroha/package.json:66]: Yes, I will remove "iroha-helpers-ts".

**Comment by petermetz** [comment]
Suggested edit:
```diff
diff --git a/packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/iroha-sign-utils.ts b/packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/iroha-sign-utils.ts
index 009db49d..ba4c2ac0 100644
--- a/packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/iroha-sign-utils.ts
+++ b/packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/iroha-sign-utils.ts
@@ -5,7 +5,7 @@
  * Signing utility functions for HL Iroha ledger.
  */
 
-import txHelper from "iroha-helpers/lib/txHelper";
+import { txHelper } from "iroha-helpers";
 import { Transaction } from "iroha-helpers/lib/proto/transaction_pb";
 
 /**

```

File: packages/cactus-plugin-ledger-connector-iroha/package.json
Comment (petermetz) [reply packages/cactus-plugin-ledger-connector-iroha/package.json:66]: @yashrajdesai: +1 to what @baziorek said!

File: packages/cactus-plugin-ledger-connector-iroha/package.json
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-iroha/package.json:65]: @yashrajdesai Please lock down the version instead of having it on auto-upgrade for the sake of reproducible builds.

**Review by petermetz — CHANGES_REQUESTED** [review]
@yashrajdesai Please resolve the merge  conflicts with upstream/main

**Comment by baziorek** [comment]
@yashrajdesai Iroha-Core-Team publisher iroha-javascript library 1.5.0: https://www.npmjs.com/package/iroha-helpers
so You can now correct the PR.

**Comment by yashrajdesai** [comment]
Sure. Will make the required changes.

**Review by baziorek — APPROVED** [review]
It looks OK, but please sign Your commit:
obraz

**Comment by baziorek** [comment]
@yashrajdesai I noticed that one of tests started failing:
https://github.com/hyperledger/cactus/actions/runs/3428326068/jobs/5712584885
but it should be easy to fix:    
obraz

**Comment by yashrajdesai** [comment]
Yes, will fix this!

**Comment by yashrajdesai** [comment]
> @yashrajdesai Changes looks good, but the question is: is the code still working?
> 
> Could You please tell us how You tested those changes? What commands/scripts/tests?

I made sure that all the existing tests of the previous version passed and modified some tests according to the new 1.5.0 version.

**Review by petermetz — CHANGES_REQUESTED** [review]
@yashrajdesai Please fix the failing tests.

**Review by petermetz — APPROVED** [review]
@yashrajdesai LGTM, thank you very much!

**Review by jagpreetsinghsasan — APPROVED** [review]

**Comment by petermetz** [comment]
This needs reviews from the other maintainers from different orgs so I'm removing my approval since Jagpreet just added one as well.

**Review by izuru0 — APPROVED** [review]
LGTM

**Pull request merged**
